packages/wpcom-template-parts/src/universal-footer-navigation/index.tsx

```diff
@@ -471,10 +473,15 @@ const UniversalNavbarFooter = ( {
 }: FooterProps ) => {
 	const localizeUrl = useLocalizeUrl();
 	const locale = useLocale();
+	const translate = useTranslate();
 	const isEnglishLocale = useIsEnglishLocale();
 	const pathNameWithoutLocale =
 		currentRoute && removeLocaleFromPathLocaleInFront( currentRoute ).slice( 1 );
-	const automatticBranding = useAutomatticBrandingNoun();
+	const [ automatticBranding, setAutomatticBranding ] = useState( { article: '', noun: '' } );
+
+	useLayoutEffect( () => {
```

Comment:
By the way `useLayoutEffect` has been causing warnings in the React 18 PR #77046. There's more info about it here: https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85

But the core idea is that since effects don't run on the server in SSR, `useLayoutEffect` can be problematic because its purpose is to update content _before_ people see it. When SSR generates a layout without using the effect, that content wouldn't include what the layout effects expects to add.

Thankfully, this isn't a problem for us because the default content is just the Automattic logo, so we have a graceful fallback.

What I did to fix the warning in #77046 was to conditionally define `useLayoutEffect` in SSR mode like this:

```js
const useIsomorphicEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect;
```

And it seems to resolve the issue. Just wanted to drop a comment here as well.

Reply:
Interestingly, it looks like this warning will be removed in a future React version: https://github.com/facebook/react/pull/26395

Reply:
This particular code would be better off if it used just `useEffect`. Without SSR, the benefit of `useLayoutEffect` would be that the version of the page with the incomplete footer would never be actually painted. The first paint would occur only after the layout effect sets the footer to the final random value.

But with SSR, the initial version is always painted, because it's a part of the initial HTML markup. `useLayoutEffect` doesn't make any difference compared to `useEffect`.